src/_tutorials/server/cmdline.md

```diff
@@ -73,156 +82,84 @@ The Dart VM supports many options.
-            stdout.write('${lineNumber++} ');
-          }
-          stdout.writeln(line);
-        }
-      } <a tabindex="0" role="button" class="highlight" data-toggle="popover" title="Exception handling" data-content='Any errors encountered in the stream are handled in the "catch" block.'>catch</a> (_) {
-        await _handleError(path);
-      }
-    }
-  }
-}
+Run the app from the command line:
 
-Future _handleError(String path) async {
-  if (<a tabindex="0" role="button" class="highlight" data-toggle="popover" title="Test the path"
-    data-content='You can get information about the file system on which your program is running. The "await" keyword causes error handling to pause until the path is available.'
-  >await FileSystemEntity.isDirectory(path)</a>) {
-    stderr.writeln('error: $path is a directory');
-  } else {
-    <a tabindex="0" role="button" class="highlight" data-toggle="popover" title="Exit code"
-      data-content="A well-behaved command-line app sets an exit code to indicate whether the program was successful."
-    >exitCode = 2;</a>
-  }
-}
+```terminal
+$ dart dcat.dart -n dcat.dart
+1 // Copyright (c) 2013, the Dart project authors. Please see the AUTHORS file
+2 // for details. All rights reserved. Use of this source code is governed by a
+3 // BSD-style license that can be found in the LICENSE file.
+...
+```
 
-</pre>
+The app displays the contents of the source code file and preceeds each line
+with a line number.
 
 ## Parsing command-line arguments
 
-The [args]({{site.pub}}/packages/args) package
-a software bundle that contains a library of Dart code, provides
-parser support for transforming raw command-line arguments
+The [args]({{site.pub}}/packages/args) package provides
+parser support for transforming command-line arguments
 into a set of options, flags, and additional values.
 Import the library as follows:
 
-{% prettify dart %}
+<?code-excerpt "misc/bin/dcat.dart" retain="package:args"?>
+```dart
 import 'package:args/args.dart';
-{% endprettify %}
+```
 
-The args library contains two classes:
+The `args` package contains these classes, among others:
 
-| Library | Description |
+| Class | Description |
 |---|---|
-| <a href="{{site.pub-api}}/args/latest/args/ArgParser-class.html" target="_blank">ArgParser</a> | A class that parses command-line arguments |
-| <a href="{{site.pub-api}}/args/latest/args/ArgResults-class.html" target="_blank">ArgResults</a> | The result of parsing command-line arguments using ArgParser. |
+| [ArgParser]({{argsAPI}}/ArgParser-class.html){:target="_blank"} | A command-line argument parser. |
+| [ArgResults]({{argsAPI}}/ArgResults-class.html){:target="_blank"} | The result of parsing command-line arguments using `ArgParser`. |
 {: .table }
 
-Let's take a look at the `dcat` sample,
-which uses ArgParser and ArgResults to parse and store its command-line arguments.
+Here is the `dcat` code that uses these classes to parse and store command-line
```

Comment:
I moved things around a bit. It seems to make more sense to show the lines of code used to process arguments before the command-line image (which refers to `ArgResults[lineNumber]` and `ArgResults.rest`).